### PR TITLE
Lt 4343 multiple withdrawal

### PR DESCRIPTION
**Pull request by gponomarev-lykke**

### CODE REVIEW

File: src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs [hunk 1]
```diff
@@ -74,23 +82,40 @@ private async Task Handle(FreezeAmountForWithdrawalCommand command, IEventPublis
...
+                    _logger.LogInformation("{Command}: AccountId {AccountId}, FreeMargin {FreeMargin}, OperationId {OperationId}",
+                        nameof(FreezeAmountForWithdrawalCommand),
+                        command.AccountId,
+                        account.GetFreeMargin(),
+                        command.OperationId
+                    );
+
+                    if (account.GetFreeMargin() >= command.Amount)
+                    {
+                        var frozenMargin = _accountUpdateService.FreezeWithdrawalMargin(command.AccountId, command.OperationId,
```
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs:96]: Don't you that the latest you acquire the lock - the better?  Why n to wrap just line 96 with lock?

File: src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs [hunk 2]
```diff
@@ -74,23 +82,40 @@ private async Task Handle(FreezeAmountForWithdrawalCommand command, IEventPublis
...
+                    
+                        _logger.LogInformation("{Command} (after freeze): AccountId {AccountId}, FrozenMargin {FrozenMargin}, OperationId {OperationId}",
+                            nameof(FreezeAmountForWithdrawalCommand),
+                            command.AccountId,
+                            frozenMargin,
+                            command.OperationId
+                        );
                     
-                    _chaosKitty.Meow(command.OperationId);
+                        _chaosKitty.Meow(command.OperationId);
```
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs:106]: Even chaosKitty.Meow and events publishing are being executed under lock.

File: src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs (hunk 1 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs:96]: account.GetFreeMargin() should be under lock to ensure proper calculation
I decided to wrap logs too so they're consistent with the if check.

File: src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs (hunk 2 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Workflow/WithdrawalCommandsHandler.cs:106]: Yep I think it makes sense to release lock a little bit earlier. I will check it.

**Review by tarurar — APPROVED** [review]

**Pull request merged**